### PR TITLE
Get trainer report positions from new position table column

### PR DESCRIPTION
**Pull request by srabraham**
This requires a corresponding ranger-clubhouse-web change, which should be merged after this one. https://github.com/burningmantech/ranger-clubhouse-web/pull/1473

There are probably rough edges here, so give me whatever feedback is relevant. I also haven't updated
any tests, but I'll do so if you point me to tests that ought to be updated.

https://github.com/burningmantech/ranger-clubhouse-web/issues/1471

### CODE REVIEW

File: app/Models/TrainingSession.php [hunk 1]
```diff
@@ -101,6 +103,14 @@ public function retrieveStudents(): array
                 $status = 'unknown';
             }
 
+            $position_short_titles = $person->person_position
+                ->filter(fn($pp) => isset($positionsToReport[$pp->position_id]))
+                ->map(fn($pp) => $positionsToReport[$pp->position_id]->short_title !== ''
```
Comment (mikeburg) [inline app/Models/TrainingSession.php:108]: The php way to check for empty strings is to use empty($value). Also, short_title can be null and empty() will also covert that case.

File: database/migrations/2022_11_11_065414_add_on_trainer_report_to_position.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up(): void
+    {
+        Schema::table('position', function (Blueprint $table) {
+            $table->boolean('on_trainer_report')->default(false)->nullable(true)->after('on_sl_report');
```
Comment (mikeburg) [inline database/migrations/2022_11_11_065414_add_on_trainer_report_to_position.php:17]: Set nullable to false. Any boolean column should be either true or false and not include a third option, null.

**Review by mikeburg — CHANGES_REQUESTED** [review]

File: app/Models/TrainingSession.php (hunk 1 above)
Comment (srabraham) [reply app/Models/TrainingSession.php:108]: nice, thanks and done

File: database/migrations/2022_11_11_065414_add_on_trainer_report_to_position.php (hunk 2 above)
Comment (srabraham) [reply database/migrations/2022_11_11_065414_add_on_trainer_report_to_position.php:17]: works for me, done. I copied that from position.on_sl_report, which is nullable

**Pull request merged**